src/scandeval/generation.py

```diff
@@ -257,12 +257,18 @@ def generate_single_iteration(
             with warnings.catch_warnings(), torch.inference_mode():
                 warnings.simplefilter("ignore", category=UserWarning)
                 inputs = batch["input_ids"].to(model.device)
+                stopping_criteria.clear()
```

Comment:
I think `generate_single_iteration` has become quite long and involved. Maybe move the code from the core loop into a function called `generate_single_batch` or something similar.